src/webserver/WebServer.jl

```diff
@@ -68,12 +68,32 @@ This will start the static HTTP server and a WebSocket server. The server runs _
 Pluto notebooks can be started from the main menu in the web browser.
 """
 function run(; kwargs...)
-    options = Configuration.from_flat_kwargs(; kwargs...)
-    run(options)
+    notebook = get(kwargs, :notebook, nothing)
+
+    # for backward compatibility - Configuation can only contain a single notebook for startup
+    notebook_for_config = notebook isa AbstractVector ? first(notebook) : notebook
+
+    options = Configuration.from_flat_kwargs(; kwargs..., notebook=notebook_for_config)
+    run(options, notebook)
+end
+
+# open notebook(s) on startup
+function run(options::Configuration.Options, notebook:: Nothing = nothing)
+    session = ServerSession(;options=options)
+    run(session)
 end
 
-function run(options::Configuration.Options)
+function run(options::Configuration.Options, notebook:: AbstractString)
     session = ServerSession(;options=options)
+    SessionActions.open(session, notebook; compiler_options=options.compiler)
```

Comment:
Do you need to manually pass along compiler_options? Are they not taken from the session?

Reply:
Right - I'll remove this.